internal/stores/objects.go

```diff
@@ -465,11 +464,20 @@ func (s *SQLStore) Put(key string, o object.Object) error {
 			}
 
 			for _, shard := range ss.Shards {
-				// Create shard.
+				// TODO: We potentially want to ignore foreign
```

Comment:
I don't understand this `TODO` - when would the FK constraint fail the contract should already exist

Reply:
It "should" but due to a race or something else it might be missing. In that case we probably don't want to fail the `Put` after an upload. Instead we could just NULL the foreign key and store it still. Then the migration code can pick up the sector.

Reply:
ok well in that case I'm not in favour of ignoring the FK because it hides potential serious database corruption issues, if that happens we want to know about and be alerted instead of it silently failing and relying on some other process to fix the corruption

I know you said "potentially" :) - just saying I would vote against.

Reply:
Ok in that case the caller (mostly autopilot) will need to guarantee that it's not producing any race conditions. Like using a contract after renewing it or similar.

Reply:
It's my understanding that contracts can be used for a little while after they're renewed. Just not for uploading, but the autopilot acquires a lock on a contract when it's renewed so a race condition should be easy to avoid #famouslastwords